query.go

```diff
@@ -65,17 +66,19 @@ func query(url, token, dns_name string, duration time.Duration) string {
 		// TODO: check that we got status code 200
 		body, err := ioutil.ReadAll(resp.Body)
 		if err != nil {
-			clog.Fatalf("Error reading body %v", err)
+			clog.Errorf("Error reading body: %v", err)
+			return ""
```

Comment:
I think the better approach would be to define:

```
ip_address := ""
```
And return this if no other answer is given in the case of all errors.

Reply:
The difference between the return of "" and the return of `ip_address` that is previously set to "" is out of my competence here. If you consider that it is important - feel free to modify my PR so it would be in accord with your preference ;)
But I still think that it would be better to return immediately in case of error and not retry (see my previous comment).